tests/datatypes/test_str.py

```diff
@@ -301,57 +301,288 @@ def test_isupper(self):
-                if 'bee'.islower():
-                    print('bee is lowercase')
-                else:
-                    print('bee is not lowercase')
-
-                # Multiple character, mixed case
-                if 'Bee'.islower():
-                    print('Bee is lowercase')
-                else:
-                    print('Bee is not lowercase')
-
-                # Multiple character, all caps with digits and punctuation
-                if 'B1E2E!'.islower():
-                    print('B1E2E! is lowercase')
-                else:
-                    print('B1E2E! is not lowercase')
-
-                # Multiple character, mixed case with digits and punctuation
-                if 'B1e2E!'.islower():
-                    print('B1e2E! is lowercase')
-                else:
-                    print('B1e2E! is not lowercase')
-
-                # Only numbers and punctuation
-                if '12!3_4'.islower():
-                    print('12!3_4 is lowercase')
-                else:
-                    print('12!3_4 is not lowercase')
-
-                print('done.')
+    def test_islower(self):
+        self.assertCodeExecution("""
+            # Single capital
+            if 'B'.islower():
+                print('B is lowercase')
+            else:
+                print('B is not lowercase')
+
+            # Single lowercase
+            if 'b'.islower():
+                print('b is lowercase')
+            else:
+                print('b is not lowercase')
+
+            # Multiple character, all caps
+            if 'BEE'.islower():
+                print('BEE is lowercase')
+            else:
+                print('BEE is not lowercase')
+
+            # Multiple character, all lowercase
+            if 'bee'.islower():
+                print('bee is lowercase')
+            else:
+                print('bee is not lowercase')
+
+            # Multiple character, mixed case
+            if 'Bee'.islower():
+                print('Bee is lowercase')
+            else:
+                print('Bee is not lowercase')
+
+            # Multiple character, all caps with digits and punctuation
+            if 'B1E2E!'.islower():
+                print('B1E2E! is lowercase')
+            else:
+                print('B1E2E! is not lowercase')
+
+            # Multiple character, mixed case with digits and punctuation
+            if 'B1e2E!'.islower():
+                print('B1e2E! is lowercase')
+            else:
+                print('B1e2E! is not lowercase')
+
+            # Only numbers and punctuation
+            if '12!3_4'.islower():
+                print('12!3_4 is lowercase')
+            else:
+                print('12!3_4 is not lowercase')
+
+            print('done.')
+        """)
+
+    def test_lstrip(self):
+        # No argument passed, strip whitespace only
+        self.assertCodeExecution("""
+            x = ' Bee'
+            print(s.lstrip())
```

Comment:
Oops...That was rather dumb, sorry. 

Now fixed the false positives & put the unit tests in one code block (and it did get remarkably faster!)

Thanks for the comment!